valet/predicates.go

```diff
@@ -355,6 +355,7 @@ func MakeIsArchived(localBase string, remoteBase string,
 				Str("expected_checksum", chk).
 				Str("checksum", obj.Checksum()).
 				Msg("not archived: checksum not confirmed")
+			return
```

Comment:
Why isn't this `!ok` check combined with the `err` check like the previous tests? It means that an err checking the checksum doesn't get a log message.

Also, earlier in this method it is possible to return `true` with an error, which I don't really like. What if the caller relies on the boolean result?

Do you apply all the standard linting on your code? I think one of them would object to a long method having lots of "blank" returns because it becomes non-obvious what is being returned. The standard thing to do would be to make each return explicit, eg. `return false, err` for the avoidance of doubt.

Reply:
On the first point, thanks - that improves it.

On the second, the caller should be checking errors. I think that a result of `false` is equally untrustworthy as `true` if there is an error (and the caller should not be rely on knowing the implementation to tell if they can ignore an error in the case of `false`)

I use `golangci-lint` (see `Makefile`). I think it's running about 10 linters. It gripes about the test code in a couple of places, but not about this. One reason I'm using named return parameters is to reduce visual noise from repeating the same values (here, 9 times).

Reply:
I think, given the choice, you may as well program it defensively and not assume the caller will do the right thing. It's a needless potential for a bug that can easily be negated with a clear `return false, err`.

The documentation for the method simply states "MakeIsArchived returns a predicate that will return true if its argument has been successfully archived from localBase to remoteBase." If I'm going to take that literally...